compiler/optimizations/gpuTransforms.cpp

```diff
@@ -1467,45 +1483,65 @@ Symbol* GpuKernel::addLocalVariable(Symbol* symInLoop) {
  *  blockDimX  = __primitive('gpu blockDim x')
  *  threadIdxX = __primitive('gpu threadIdx x')
  *  t0 = varBlockIdxX * varBlockDimX
- *  t1 = t0 + threadIdxX
- *  index = t1 + lowerBound
+ *  t1 = t0 + threadIdxX      // aka `global thread idx`
+ *  index = t1 + lowerBound   // aka `calculated thread idx`
+ *
+ *  If we have gpu.itersPerThread, instead generates:
+ *
+ *  index = lowerBound + t1 * itersPerThread
  *
  *  Also adds the loopIndex->index to the copyMap_
  **/
 void GpuKernel::generateIndexComputation() {
   std::vector<Symbol*>::size_type numIndices = gpuLoop.loopIndices().size();
```

Comment:
Is this ever not `1` today? I'm surprised that moving the `varBlockIdxX` assignments out of the loop was not done earlier (since `blockIdxX` will not change within a single thread) but I can see it being excusable if it's the loop only completes one iteration.

Reply:
I don't know. I assumed that numIndices > 1 for a zippered loop, however did not check that.